internal/broker/broker_test.go

```diff
@@ -827,6 +831,147 @@ func TestConcurrentIsAuthenticated(t *testing.T) {
 	}
 }
 
+func TestIsAuthenticatedAllowedUsersConfig(t *testing.T) {
+	t.Parallel()
+	tests := map[string]struct {
+		allowedUsers          map[string]struct{}
+		owner                 string
+		ownerAllowed          bool
+		allUsersAllowed       bool
+		firstUserBecomesOwner bool
+
+		allowedUsernames   []string
+		unallowedUsernames []string
+	}{
+		"All_users_are_allowed": {
+			allUsersAllowed:  true,
+			allowedUsernames: []string{"u1", "u2"},
+		},
+		"Only_owner_allowed": {
+			ownerAllowed:       true,
+			owner:              "machine_owner",
+			allowedUsernames:   []string{"machine_owner"},
+			unallowedUsernames: []string{"u1", "u2"},
+		},
+		"No_users_allowed": {
+			ownerAllowed:       true,
+			owner:              "",
+			unallowedUsernames: []string{"u1", "u2", "machine_owner", "u3"},
+		},
+		"Only_unset_owner_allowed": {
+			ownerAllowed:          true,
+			firstUserBecomesOwner: true,
+			allowedUsernames:      []string{"random_user"},
+			unallowedUsernames:    []string{"u1", "u2"},
+		},
+		"Specific_users_allowed": {
+			allowedUsers:       map[string]struct{}{"u1": {}, "u2": {}},
+			allowedUsernames:   []string{"u1", "u2"},
+			unallowedUsernames: []string{"u3"},
+		},
+		"Specific_users_and_owner": {
+			allowedUsers:       map[string]struct{}{"u1": {}, "u2": {}},
+			ownerAllowed:       true,
+			owner:              "machine_owner",
+			allowedUsernames:   []string{"u1", "u2", "machine_owner"},
+			unallowedUsernames: []string{"u3"},
+		},
+		"Owner_is_disabled_even_when_registered": {
```

Comment:
Maybe a bit easier to understand

```suggestion
		"Owner_is_set_but_not_allowed": {
```

Reply:
We can also name it consistently with the other `No_users_allowed` test cases and group them together: 66996aa01d9f7b6669766b16e374908fa80e9f91